src/__tests__/utils.test.ts

```diff
@@ -0,0 +1,29 @@
+import type { Repository } from '../models'
+
+import { mapRepositoryToQueryParams } from '~/utils'
+
+describe('mapRepositoryToQueryParams util', () => {
+  it('should map an undefined repository', () => {
+    const result = mapRepositoryToQueryParams()
+
+    expect(result).toEqual({ owner: '', repo: '' })
+  })
+
+  it('should map a full repository', () => {
+    const result = mapRepositoryToQueryParams({
+      owner: { login: 'foo' },
+      name: 'bar',
+    } as Repository)
+
+    expect(result).toEqual({ owner: 'foo', repo: 'bar' })
+  })
+
+  it('should map a partial repository', () => {
+    const result = mapRepositoryToQueryParams({
+      owner: {},
+      name: 'bar',
+    } as Repository)
+
+    expect(result).toEqual({ owner: '', repo: 'bar' })
```

Comment:
The input data for the tests is similar, but not 100% the same. I'm wondering if we could simplify the setup steps by putting the data in a table?

Example from Jest docs:

```js
test.each`
  a    | b    | expected
  ${1} | ${1} | ${2}
  ${1} | ${2} | ${3}
  ${2} | ${1} | ${3}
`('returns $expected when $a is added $b', ({a, b, expected}) => {
  expect(a + b).toBe(expected);
});
```

> Use `test.each` if you keep duplicating the same test with different data. `test.each` allows you to write the test once and pass data in.

https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
Yes, we definitely could. I didn't want to put that effort into tests for this PR. I'll manage this in the next round of changes, where I'll add tests for the whole file.